pkg/bpf/map_linux_test.go

```diff
@@ -51,64 +43,55 @@ func (k *TestKey) New() MapKey    { return &TestKey{} }
 func (v *TestValue) String() string { return fmt.Sprintf("value=%d", v.Value) }
 func (v *TestValue) New() MapValue  { return &TestValue{} }
 
-var _ = Suite(&BPFPrivilegedTestSuite{})
-
-func (s *BPFPrivilegedTestSuite) SetUpSuite(c *C) {
-	testutils.PrivilegedTest(c)
+func setup(tb testing.TB) {
+	testutils.PrivilegedTest(tb)
 
 	CheckOrMountFS("")
 
 	if err := rlimit.RemoveMemlock(); err != nil {
-		c.Fatal(err)
+		tb.Fatal(err)
 	}
 
+	testMap = NewMap("cilium_test",
+		ebpf.Hash,
+		&TestKey{},
+		&TestValue{},
+		maxEntries,
+		BPF_F_NO_PREALLOC,
+	).WithCache()
+
 	if err := testMap.OpenOrCreate(); err != nil {
```

Comment:
No real need to pin this map afaik, let's change it:
```suggestion
	if err := testMap.CreateUnpinned(); err != nil {
```

This also allows us to get rid of the `os.Remove()` in tb.Cleanup.

Reply:
Thanks, this should be incorporated in https://github.com/cilium/cilium/pull/32220